controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -1415,7 +1433,9 @@ func (r *BareMetalHostReconciler) manageHostPower(prov provisioner.Provisioner,
 		"reboot process", desiredPowerOnState != info.host.Spec.Online)
 
 	if desiredPowerOnState {
-		if isProvisioned {
+		// If the host is provisioned/externallyProvisioned or if its in
+		// the available state after deprovisioning.
+		if isProvisioned || isAvailable {
```

Comment:
nit: I'm not sure we can enter this function in any other state.

Reply:
Thinking more about it, I don't believe using Available is the right thing to do. The image won't be disconnected until a reboot, which will happen.. who knows when. I think we should start deprovisioning with deleting a data image.

Reply:
You mean just deleting the data image and not detaching it, right ?